### PR TITLE
3757 : Refactored test_introducer in web tests

### PR DESCRIPTION
**Pull request by Fenn-CS**
Refactored `test_introducer.py` in `test/web` module to either user `SyncTestCase` or `AsyncTestCase`.

Ticket : https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3757

### CODE REVIEW

File: src/allmydata/test/web/test_introducer.py
Comment (exarkun) [inline src/allmydata/test/web/test_introducer.py:239]: The convention of an identifier prefixed with `_` indicates something private or subject to change or which is meant to be an implementation detail.  In general, Tahoe code can't use private APIs from other libraries. Specifically, Tahoe can't use this `_matcher` attribute.

The intended way to use `succeeded` is demonstrated by https://github.com/tahoe-lafs/tahoe-lafs/blob/93fbfb61a8335428ad97d592842a5db78d26e30e/src/allmydata/test/test_eliotutil.py#L252-L260

The assertion here could probably be something like:

```
        response = render(resource, {b"t": [b"json"]})
        expected = {
            u"subscription_summary": {"arbitrary": 2},
            u"announcement_summary": {"arbitrary": 1},
        }
        self.assertThat(response, succeeded(AfterPreprocessing(json.loads, Equals(expected))))
```

File: src/allmydata/test/web/test_introducer.py
Comment (exarkun) [inline src/allmydata/test/web/test_introducer.py:244]: The `result` attribute of `Deferred` is _not_ marked as private by the `_` prefix convention ... but it is not good practice to use it, either.  Fortunately I think correct use of `succeeded` will remove the need for this attribute.

File: src/allmydata/test/web/test_introducer.py
Comment (exarkun) [inline src/allmydata/test/web/test_introducer.py:115]: This extra blank line has trailing whitespace.  The whole line should be deleted.

File: src/allmydata/test/web/test_introducer.py
Comment (exarkun) [inline src/allmydata/test/web/test_introducer.py:236]: This line has trailing whitespace added that should be removed.

File: src/allmydata/test/web/test_introducer.py
Comment (exarkun) [inline src/allmydata/test/web/test_introducer.py:18]: This import is no longer used and can be deleted.

**Review by exarkun — CHANGES_REQUESTED** [review]
Thanks.  I left some commentary inline.  This looks like a good start.

File: src/allmydata/test/web/test_introducer.py
Comment (Fenn-CS) [reply src/allmydata/test/web/test_introducer.py:239]: I  thought as much about this being somewhat weird but never knew exactly what interface to use that would pick our the correct props to make the comparison succeed. Thank you!

**Review by meejah — APPROVED** [review]
Looking really good!

I'm not asking for changes here, but especially for shorter tests like this one I think we (or at least "I") have been indenting things like this:

```
self.assertThat(
    render(resource, {b"t": [b"json"]}),
    succeeded(
        AfterPreprocessing(
            json.loads,
            Equals({
                u"subscription_summary": {"arbitrary": 2},
                u"announcement_summary": {"arbitrary": 1},
            })
        )
    )
)
```

I'm actually curious if you think that counts as "readable", or if it's just because I'm now used to it ;)

If nothing else, I do like:

```
self.assertThat(
    <something>,
    CollectionOfMatchers(),
)
```

...to separate the "test thing" (`<something>`) from the "expected results part" (i.e. all the testtools matchers). I do think it took me a while to get around the "matcher" style of test-tools, but it can be fairly readable (even if the writing can be a little harder, due to having to look up matchers).

p.s. I know I said "approve" but maybe @exarkun wishes to followup .. and CI has to be green before we merge things...

**Comment by Fenn-CS** [comment]
@meejah Yeah, I think that's more readable. I ran the linting I found the `tox` config to see if would complain or refactor that, did not! Could quickly push a new commit with that. If we would use it in other places we might as well just do it here.

**Comment by May-Lee** [comment]
@exarkun could you approve requested changes and merge when ready please?

**Pull request merged**
